# Type: fix
# Subtype: I3
# English Summary: Added check for non-password mode
# Chinese Summary: 添加对非密码模式的判断
# Entry Number: 30

# === Git Diff Content ===

diff --git a/gdx/src/com/badlogic/gdx/scenes/scene2d/ui/TextField.java b/gdx/src/com/badlogic/gdx/scenes/scene2d/ui/TextField.java
index f11c669..8830a8b 100644
--- a/gdx/src/com/badlogic/gdx/scenes/scene2d/ui/TextField.java
+++ b/gdx/src/com/badlogic/gdx/scenes/scene2d/ui/TextField.java
@@ -558,7 +558,7 @@ public class TextField extends Widget {
 
 	/** Copies the contents of this TextField to the {@link Clipboard} implementation set on this TextField. */
 	public void copy () {
-		if (hasSelection) {
+		if (hasSelection && !passwordMode) {
 			int minIndex = Math.min(cursor, selectionStart);
 			int maxIndex = Math.max(cursor, selectionStart);
 			clipboard.setContents(text.substring(minIndex, maxIndex));
@@ -568,7 +568,7 @@ public class TextField extends Widget {
 	/** Copies the selected contents of this TextField to the {@link Clipboard} implementation set on this TextField, then removes
 	 * it. */
 	public void cut () {
-		if (hasSelection) {
+		if (hasSelection && !passwordMode) {
 			copy();
 			delete();
 		}
